### ISSUE TITLE
Suggest alternative HTML structure for listbox example with grouped options

### ISSUE THREAD

**Issue by backwardok** [w3c/aria-practices#2800]
While looking at the [listbox example with grouped options](https://www.w3.org/WAI/ARIA/apg/patterns/listbox/examples/listbox-grouped/), I noticed that the markup for the groups creates a flat hierarchy across the groups, with each label for the group being a list item of that containing list like so (simplified):
```html
<div role="listbox">
  <ul role="group" aria-labelledby="groupLabel">
    <li role="presentation" id="groupLabel">
      Group label
    </li>
    <li role="option">Group item</li>
    ...
  </ul>
</div>
```

I wonder if it'd be worth adjusting the example markup so that the list has the same hierarchy as what's shown visually, the way that nested lists work, like so (simplified):
```html
<ul role="listbox">
  <li role="group" aria-labelledby="groupLabel">
    <span id="groupLabel">Group label</span>
    <ul role="presentation">
      <li role="option">Group item</li>
      ...
    </ul>
  </li>
</ul>
```

I created a [codepen fork off of the one in the example](https://codepen.io/backwardok/pen/wvRemxj/903d67c08e796bb352d167fd045c32e2) with the version I suggested at the top and the original on the bottom. 

In checking the accessibility tree in Firefox, Chrome, and Safari, the accessibility nodes and hierarchy appear to be the same across the version I made and the version in the example.

**Comment by backwardok** [comment]
(if you read this before this comment, I was in the middle of editing and tabbed forward to the save button by accident and hit enter. Description has since been updated)

**Comment by backwardok** [comment]
Something I forgot to mention - having the outer element as a `<ul>` also follows the same pattern as a non-grouped listbox (where it's recommended to have a `<ul>`). This also means that if we were to have a mix of non-grouped items and grouped items, they'd all fit within the same markup pattern

**Comment by backwardok** [comment]
Oh also one more thing! The idea that this is one list with 11 items is more in line with using a singular outer `<ul>` with nested lists inside it, vs the existing APG example which has a div with 3 lists inside it

**Comment by mcking65** [comment]
@backwardok 

What browsers expose in their accessibility trees ends up being identical regardless of which HTML structure you use. The differences in how posinset and setsize are calculated across browsers is due to the fact that the ARIA spec does not include normative requirements for that calculation. Nothing about how the widget is built will change that.

I remember discussing that calculation in the ARIA WG back in the 1.1 timeframe. There were strong differences of opinion. We probably decided to close the issue without resolution, which implicitly defers it to implementers. That's not great for interoperability. That said, there are far more consequential issues that need addressing, so I'm not sure that raising the setsize calculation for option groups again would change anything in the foreseeable future.

The bottom line is that you can use either HTML list structure as well as many other possible combinations of elements as well.

I'm going to close this issue. However, if you feel it needs further discussion, please feel free to re-open it.

**Comment by backwardok** [comment]
Hi @mcking65! I'm not sure I understand what your comment has to do with my comment? My thought was that using a similar structure to the simple listbox makes it easier in code to extend the concept to having subgroups, while still having top level options. And that using a single list with sublists fits more with the structure of the listbox. 

I don't feel super strongly about this needing to be updated to the markup I shared, but thought I'd suggest it since it was something that came up in my own implementation.
